### PR TITLE
[DOC] update haxby full analysis example

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Relates to #6265

Use of AI tools:
- [ ] LLM tools were used to generate at least part of the content of this pull-request.
- [x] No LLM tools were used to generate any part of the content of this pull-request.

Changes proposed in this pull request:

- clean up example
- silence warnings
- properly standardize the data

### CODE REVIEW

File: examples/02_decoding/plot_haxby_full_analysis.py [hunk 1]
```diff
@@ -63,84 +64,83 @@
...
 mask_scores = {}
 mask_chance_scores = {}
 
 for mask_name in mask_names:
     print(f"Working on {mask_name}")
-    # For decoding, standardizing is often very important
+
     mask_filename = haxby_dataset[mask_name][0]
-    masker = NiftiMasker(mask_img=mask_filename, verbose=1)
+    masker = NiftiMasker(mask_img=mask_filename, standardize="zscore_sample")
```
Comment (Remi-Gau) [inline examples/02_decoding/plot_haxby_full_analysis.py]: we were not standardizing the data anymore (despite the comment): 
older versions of the doc were

**Comment by Remi-Gau** [comment]
Cannot yet identify what's the cause of having lower performance than in 0.13.1

See results of 0.13.1

File: examples/02_decoding/plot_haxby_full_analysis.py [hunk 2]
```diff
@@ -61,88 +62,79 @@
...
 
 for mask_name in mask_names:
-    print(f"Working on {mask_name}")
-    # For decoding, standardizing is often very important
+    print(f"\nWorking on {mask_name}")
+
     mask_filename = haxby_dataset[mask_name][0]
-    masker = NiftiMasker(
-        mask_img=mask_filename, verbose=1, standardize="zscore_sample"
-    )
```
Comment (Remi-Gau) [inline examples/02_decoding/plot_haxby_full_analysis.py:91]: chose to pass directly the mask to the decoders and apply the standardization there (it's the default anyway), rather than instantiating a masker first.

File: examples/02_decoding/plot_haxby_full_analysis.py [hunk 3]
```diff
@@ -61,88 +62,79 @@
 # -------------------------------
 #
 # The classifier used here is a support vector classifier (svc).
```
Comment (hndgzkn) [inline examples/02_decoding/plot_haxby_full_analysis.py]:
```suggestion
# The classifier used here is a Support Vector Classifier (SVC).
```

File: examples/02_decoding/plot_haxby_full_analysis.py [hunk 4]
```diff
@@ -61,88 +62,79 @@
...
-# We use
-# :class:`~nilearn.decoding.Decoder` and specify the classifier.
-import numpy as np
+# We use :class:`~nilearn.decoding.Decoder`
+# with a ``svc_l1`` estimator because it is intra subject.
+# The mask of the region of interest is passed directly to the Decoder.
+#
+# We will be doing a 'leave one run out' for cross validation
+# by using LeaveOneGroupOut and passing the run labels at fit time.
```
Comment (hndgzkn) [inline examples/02_decoding/plot_haxby_full_analysis.py]:
```suggestion
# by using :class:`sklearn.model_selection.LeaveOneGroupOut` and passing the run labels at fit time.
```

**Review by hndgzkn — APPROVED** [review]
Minor suggestions, otherwise LGTM.

File: examples/02_decoding/plot_haxby_full_analysis.py
Comment (hndgzkn) [inline examples/02_decoding/plot_haxby_full_analysis.py]:
```suggestion
ROI-based decoding analysis in the Haxby dataset
```

Comment (hndgzkn) [reply examples/02_decoding/plot_haxby_full_analysis.py]: It seems that the dataset is referred in general as `Haxby dataset`.

**Comment by Remi-Gau** [comment]
- see https://github.com/nilearn/nilearn/issues/6525 for why we have a drop in performance

File: examples/02_decoding/plot_haxby_full_analysis.py [hunk 5]
```diff
@@ -6,52 +6,53 @@
...
+    "ignore", message="The provided image has no sform in its header."
+)
+warnings.filterwarnings(
+    "ignore", message="The decoding model will be trained only on"
+)
 
 # %%
 # Load and prepare the data
 # -------------------------
+# We fetch the data a single subject for analysis.
```
Comment (hndgzkn) [inline examples/02_decoding/plot_haxby_full_analysis.py]:
```suggestion
# We fetch the data of a single subject for analysis.
```

File: examples/02_decoding/plot_haxby_full_analysis.py [hunk 6]
```diff
@@ -1,57 +1,58 @@
 """
-ROI-based decoding analysis in Haxby et al. dataset
+ROI-based decoding analysis in the Haxby dataset
 ===================================================
```
Comment (Remi-Gau) [inline examples/02_decoding/plot_haxby_full_analysis.py]:
```suggestion
ROI-based decoding analysis in the Haxby dataset
================================================
```

File: examples/02_decoding/plot_haxby_full_analysis.py [hunk 7]
```diff
@@ -60,89 +61,80 @@
...
-# We use
-# :class:`~nilearn.decoding.Decoder` and specify the classifier.
-import numpy as np
+# The classifier used here is a Support Vector Classifier (SVC).
+# We use :class:`~nilearn.decoding.Decoder`
+# with a ``svc_l1`` estimator because it is intra subject.
+# The mask of the region of interest is passed directly to the Decoder.
+#
+# We will be doing a 'leave one run out' for cross validation
+# by using :class:`sklearn.model_selection.LeaveOneGroupOut` and passing the run labels at fit time.
```
Comment (Remi-Gau) [inline examples/02_decoding/plot_haxby_full_analysis.py]:
```suggestion
# by using :class:`sklearn.model_selection.LeaveOneGroupOut`
# and passing the run labels at fit time.
```
